### PR TITLE
Add a CSV output format.

### PR DESCRIPTION
**Pull request by cmccoy**

### CODE REVIEW

File: tests/publisher_test.py [hunk 1]
```diff
@@ -335,3 +336,38 @@ def testDiskLegacyDiskType(self):
...
+    samples = [{'test': 'testb', 'metric': '1', 'value': 1.0, 'unit': 'MB',
+                'metadata': {}},
+               {'test': 'testb', 'metric': '2', 'value': 14.0, 'unit': 'MB',
+                'metadata': {}},
+               {'test': 'testa', 'metric': '3', 'value': 47.0, 'unit': 'us',
+                'metadata': {}}]
+    instance.PublishSamples(samples)
+    self.tf.seek(0)
+    rows = list(csv.DictReader(self.tf))
+    self.assertItemsEqual(['1', '2', '3'], [i['metric'] for i in rows])
```
Comment (noahl) [inline tests/publisher_test.py]: Why not just test `assertEqual(rows, samples)`? Does something break?

**Comment by noahl** [comment]
Other than my one comment, LGTM.

**Comment by lkrish** [comment]
LGTM

File: tests/publisher_test.py (hunk 1 above)
Comment (cmccoy) [inline tests/publisher_test.py]: They won't be the same: after conversion to CSV, the metadata field will be gone.

**Pull request merged**